types/bool_test.go

```diff
@@ -269,3 +269,46 @@ func TestBoolEqual(t *testing.T) {
 		})
 	}
 }
+
+func TestBoolString(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		input       Bool
+		expectation string
+	}
+	tests := map[string]testCase{
+		"true": {
+			input:       Bool{Value: true},
+			expectation: "true",
+		},
+		"false": {
+			input:       Bool{Value: false},
+			expectation: "false",
+		},
+		"unknown": {
+			input:       Bool{Unknown: true},
+			expectation: "<unknown>",
+		},
+		"null": {
+			input:       Bool{Null: true},
+			expectation: "<null>",
+		},
+		"default-false": {
+			input:       Bool{},
+			expectation: "false",
+		},
+	}
+
+	for name, test := range tests {
+		name, test := name, test
```

Comment:
Nit: Do we need this declaration? It may be redundant

Reply:
Same comment for the rest of the tests

Reply:
it is one thing of Go that I really don't like: it's about how closures keep a reference to the variable and not the value.

This means that each closure would refer to the last value `name` and `test` got during the loop, not the one that was in `name` and `test` when the specific closure got created.

So, the only way to circumvent it is to just create 2 variables local to the loop.

To show the effects, https://go.dev/play/p/JFh-5r_DgnS